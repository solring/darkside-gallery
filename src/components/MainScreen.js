import React, { useState } from 'react'

import Sidebar from './Sidebar'
import Gallery from './Gallery'

import Switch from './Switch'
import ThemeContext, { themes } from '../context/ThemeContext'

function MainScreen(props) {

  const [dark, setDark] = useState(false)

  return (
    <ThemeContext.Provider value={dark ? themes.dark : themes.default}>
      <div className="row g-0">
        <div className="col-md-3">
          <Sidebar footer={
            <Switch
              val={dark}
              setVal={setDark}
              style={{ zIndex: 1070 }}
            />
          }/>
        </div>
        <div className="col-md-9">
          <Gallery />
        </div>
      </div>
    </ThemeContext.Provider>
  )
}

export default MainScreen

