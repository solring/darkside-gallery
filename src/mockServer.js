import { createServer } from 'miragejs'
import * as data from './utils/mockdata'

const API_FETCH_ARTICLES = "/api/articles/"
const API_FETCH_CATS = "/api/categories/"

const tags = Array(7).fill(0).map((e, i) => `tag${i}`)

export default function mockServer() {
  return createServer({

    routes() {
      this.get(`${API_FETCH_ARTICLES}:category`, (schema, request) => {

        const category = request.params.category
        const {start, length} = request.queryParams;

        console.log(`GET ARTICLE: cate: ${category}`);

        return {
          data: {
            list: data.genItemsStatic(length),
            next: start + length,
            tags: tags
          }
        }
      })

      this.get( API_FETCH_ARTICLES , (schema, request) => {

        const {start, length} = request.queryParams;

        console.log(`GET ARTICLE: all`);

        return {
          data: {
            list: data.genItemsStatic(length),
            next: start + length,
            tags: tags
          }
        }
      })

      this.get( API_FETCH_CATS , (schema, request) => {
        console.log(`GET CATS:`);
        return data.mockCats;
      })
    }
  });
}
