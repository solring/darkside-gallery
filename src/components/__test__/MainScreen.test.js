import React from 'react'
import { render, screen, waitFor, fireEvent, getByRole, getAllByTestId, waitForElementToBeRemoved } from '@testing-library/react'

import MainScreen from '../MainScreen'

import { Provider } from 'react-redux'
import createStore from '../../store'

// mock server
import mockServer from '../../mockServer'

let store = null
let server = null

beforeAll(() => {
  server = mockServer()
})

afterAll(() => {
  server.shutdown()
})

beforeEach(() => {
  store = createStore()
})

it('General tests: phone layout', async () => {
  const { container } = render(
    <Provider store={store}>
      <MainScreen />
    </Provider>
  )

  // Wait for fetch to complete
  await waitFor(() => screen.findByTestId('navTabs'), {timeout: 10000})
  await waitFor(() => screen.findByTestId('picGrid'), {timeout: 10000})

  // snapshot
  expect(container.firstChild).toMatchSnapshot()

  // Click tabs
  const tabs = screen.getByTestId('navTabs')
  const tab1 = getByRole(tabs, 'button', {name: "fan-art"})
  const tab2 = getByRole(tabs, 'button', {name: "illustration"})
  const tab3 = getByRole(tabs, 'button', {name: "design"})

  expect(screen.getAllByRole('button', {name: /#.*?/})).toHaveLength(9)

  fireEvent.click(tab1)
  // TODO: check articles

  fireEvent.click(tab2)
  // TODO: check articles

  fireEvent.click(tab3)
  // TODO: check articles
})

it('General tests: normal layout', async () => {

  const mockMedia = (query) => {
    return {
      matches: false,
      media: query,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    }
  }
  global.matchMedia = mockMedia

  const { container } = render(
    <Provider store={store}>
      <MainScreen />
    </Provider>
  )

  // Wait for fetch to complete
  await waitFor(() => screen.findByTestId('navTabs'), {timeout: 10000})
  await waitFor(() => screen.findByTestId('picGrid'), {timeout: 10000})

  // snapshot
  expect(container.firstChild).toMatchSnapshot()

  // Click tabs
  const tabs = screen.getByTestId('navTabs')
  const tab1 = getByRole(tabs, 'button', {name: "fan-art"})
  const tab2 = getByRole(tabs, 'button', {name: "illustration"})
  const tab3 = getByRole(tabs, 'button', {name: "design"})

  fireEvent.click(tab1)
  const tag = await screen.findByText(/Golden Kamuy/)
  expect(screen.getAllByRole('button', {name: /#.*?/})).toHaveLength(6)

  fireEvent.click(tab2)

  fireEvent.click(tab3)
  expect(await screen.findByText(/Poster/)).toBeInTheDocument()
  expect(screen.getAllByRole('button', {name: /#.*?/})).toHaveLength(3)
})