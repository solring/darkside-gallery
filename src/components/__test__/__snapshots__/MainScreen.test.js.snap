// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General tests: normal layout 1`] = `
<div
  class="row g-0 vh-100"
>
  <div
    class="col-md-3 h-100"
  >
    <header
      class="sidebar theme-transition"
      style="background: rgb(255, 255, 255);"
    >
      <div
        class="h-100 d-flex flex-column justify-content-between pt-4"
      >
        <div
          class="order-md-2"
        >
          <div>
            <div
              style="z-index: 1070;"
            >
              <div
                class="form-check form-switch d-flex align-items-center"
              >
                <input
                  class="form-check-input mt-0 me-2"
                  id="switch_undefined"
                  style="width: 4rem; height: 2rem;"
                  type="checkbox"
                  value="false"
                />
                <label
                  class="form-check-label theme-transition lh-1"
                  for="switch_undefined"
                >
                  <span
                    class="material-icons icon lg undefined"
                  >
                    dark_mode
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="order-md-1"
        >
          <h1
            class="visually-hidden"
          >
            Welcome to My Rabbit Hole
          </h1>
          <svg
            viewBox="0 0 250 174"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <lineargradient
                gradientTransform="rotate(-15)"
                id="themeGradient"
              >
                <stop
                  offset="0%"
                  stop-color="gradientColor1"
                />
                <stop
                  offset="40%"
                  stop-color="gradientColor2"
                />
              </lineargradient>
            </defs>
            <text
              class="font-title"
              fill="url(#themeGradient)"
              x="0"
              y="0"
            >
              <tspan
                dy="58"
                font-size="54"
                x="0"
              >
                Welcome
              </tspan>
              <tspan
                dy="58"
                font-size="54"
                x="0"
              >
                to my
              </tspan>
              <tspan
                dy="58"
                font-size="54"
                x="0"
              >
                Rabbit Hole.
              </tspan>
            </text>
          </svg>
          <h2
            class="h4 text-muted mt-5"
          >
            Personal Gallery of Solring.
          </h2>
          <div
            class="mt-4 d-inline-block mb-3"
          >
            <nav
              class="position-relative icon-socialmedia bg-theme-gradient theme-transition"
            >
              <ul
                class="position-absolute top-0 left-0"
              >
                <li
                  class="d-inline-block me-2"
                >
                  <a
                    class="btn"
                    href="https://github.com/solring"
                  >
                    <span
                      class="visually-hidden"
                    >
                      github
                    </span>
                  </a>
                </li>
                <li
                  class="d-inline-block me-2"
                >
                  <a
                    class="btn"
                    href="https://sunburnt-goldfish.tumblr.com"
                  >
                    <span
                      class="visually-hidden"
                    >
                      tumblr
                    </span>
                  </a>
                </li>
                <li
                  class="d-inline-block me-2"
                >
                  <a
                    class="btn"
                    href="https://twitter.com/burntgoldfish"
                  >
                    <span
                      class="visually-hidden"
                    >
                      twitter
                    </span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    class="col-md-9 h-100"
  >
    <div
      class="h-100"
    >
      <div
        class="undefined theme-transition"
        style="overflow-x: hidden; overflow-y: scroll; height: 100%;"
      >
        <nav
          class="navtabs"
          data-scroll="false"
          data-testid="navTabs"
        >
          <ul
            class="inner"
            data-open="false"
            role="tablist"
          >
            <li
              class="d-inline-block me-3 itemWrap"
              data-active="false"
              role="tab"
            >
              <button
                class="
              hvr-underline-from-center
              item
              
            "
              >
                <h3
                  class="h5"
                >
                  fan-art
                </h3>
              </button>
            </li>
            <li
              class="d-inline-block me-3 itemWrap"
              data-active="false"
              role="tab"
            >
              <button
                class="
              hvr-underline-from-center
              item
              
            "
              >
                <h3
                  class="h5"
                >
                  illustration
                </h3>
              </button>
            </li>
            <li
              class="d-inline-block me-3 itemWrap"
              data-active="false"
              role="tab"
            >
              <button
                class="
              hvr-underline-from-center
              item
              
            "
              >
                <h3
                  class="h5"
                >
                  design
                </h3>
              </button>
            </li>
          </ul>
          <div
            aria-expanded="false"
            class="bg-light collapse"
            role="tabpanel"
          >
            <div
              class="container"
            >
              <div
                class="d-flex flex-wrap justify-content-center py-3"
              />
            </div>
          </div>
        </nav>
        <div
          class=""
          data-testid="picGrid"
        >
          <div
            class="row g-3 my-3 mx-sm-3"
          >
            <div
              class="col"
            >
              <div>
                <button
                  class="react-reveal btn d-block"
                  style="opacity: 0;"
                >
                  <article
                    class="shadow-btn card mb-3"
                  >
                    <img
                      alt="Title"
                      src="emilywang.jpg"
                    />
                    <div
                      class="overlay"
                      data-longpress="false"
                    >
                      <div
                        class="p-3 text"
                      >
                        <h4>
                          Title
                        </h4>
                        <p
                          class="subtitle"
                        >
                          this is a test
                        </p>
                        <div
                          class="text-wrap"
                        >
                          <small
                            class="me-1"
                          >
                            #
                            tag1
                          </small>
                          <small
                            class="me-1"
                          >
                            #
                            tag2
                          </small>
                          <small
                            class="me-1"
                          >
                            #
                            tag3
                          </small>
                        </div>
                      </div>
                    </div>
                  </article>
                </button>
              </div>
            </div>
            <div
              class="col"
            />
            <div
              class="col"
            />
            <div
              class="col"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`General tests: phone layout 1`] = `
<div
  class="d-flex flex-column disable-select vh-100"
>
  <div
    class="flex-shrink-0 flex-grow-0"
  >
    <header
      class="sidebar theme-transition"
      style="background: rgb(255, 255, 255);"
    >
      <div
        class="collapse show"
      >
        <div
          class="h-100 d-flex flex-column justify-content-between pt-4"
        >
          <div
            class="order-md-2"
          >
            <div>
              <div
                style="z-index: 1070;"
              >
                <div
                  class="form-check form-switch d-flex align-items-center"
                >
                  <input
                    class="form-check-input mt-0 me-2"
                    id="switch_undefined"
                    style="width: 4rem; height: 2rem;"
                    type="checkbox"
                    value="false"
                  />
                  <label
                    class="form-check-label theme-transition lh-1"
                    for="switch_undefined"
                  >
                    <span
                      class="material-icons icon lg undefined"
                    >
                      dark_mode
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="order-md-1"
          >
            <h1
              class="visually-hidden"
            >
              Welcome to My Rabbit Hole
            </h1>
            <svg
              viewBox="0 0 250 174"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <lineargradient
                  gradientTransform="rotate(-15)"
                  id="themeGradient"
                >
                  <stop
                    offset="0%"
                    stop-color="gradientColor1"
                  />
                  <stop
                    offset="40%"
                    stop-color="gradientColor2"
                  />
                </lineargradient>
              </defs>
              <text
                class="font-title"
                fill="url(#themeGradient)"
                x="0"
                y="0"
              >
                <tspan
                  dy="58"
                  font-size="54"
                  x="0"
                >
                  Welcome
                </tspan>
                <tspan
                  dy="58"
                  font-size="54"
                  x="0"
                >
                  to my
                </tspan>
                <tspan
                  dy="58"
                  font-size="54"
                  x="0"
                >
                  Rabbit Hole.
                </tspan>
              </text>
            </svg>
            <h2
              class="h4 text-muted mt-5"
            >
              Personal Gallery of Solring.
            </h2>
            <div
              class="mt-4 d-inline-block mb-3"
            >
              <nav
                class="position-relative icon-socialmedia bg-theme-gradient theme-transition"
              >
                <ul
                  class="position-absolute top-0 left-0"
                >
                  <li
                    class="d-inline-block me-2"
                  >
                    <a
                      class="btn"
                      href="https://github.com/solring"
                    >
                      <span
                        class="visually-hidden"
                      >
                        github
                      </span>
                    </a>
                  </li>
                  <li
                    class="d-inline-block me-2"
                  >
                    <a
                      class="btn"
                      href="https://sunburnt-goldfish.tumblr.com"
                    >
                      <span
                        class="visually-hidden"
                      >
                        tumblr
                      </span>
                    </a>
                  </li>
                  <li
                    class="d-inline-block me-2"
                  >
                    <a
                      class="btn"
                      href="https://twitter.com/burntgoldfish"
                    >
                      <span
                        class="visually-hidden"
                      >
                        twitter
                      </span>
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hiddenButton"
      >
        <span
          class="visually-hidden"
        >
          toggle sidebar
        </span>
      </div>
    </header>
  </div>
  <div
    class="flex-shrink-1 flex-grow-1 overflow-hidden"
  >
    <div
      class="h-100"
    >
      <div
        class="undefined theme-transition"
        style="overflow-x: hidden; overflow-y: scroll; height: 100%;"
      >
        <div
          class="collapse collapse"
          data-testid="fs-collapse"
        >
           
          <div
            class="wrapper"
          >
            <div>
              <div
                class="header bg-light"
              >
                <div>
                  <span
                    class="text-uppercase h3"
                  >
                    <span
                      class="material-icons icon md align-middle me-2"
                    >
                      filter_list
                    </span>
                    Filter
                  </span>
                </div>
                <button
                  class="btn btn-link text-dark"
                  type="button"
                >
                  <span
                    class="material-icons icon-lg"
                  >
                    close
                  </span>
                </button>
              </div>
              <div
                class="content"
              >
                <div
                  class="wrapper shadow-down"
                  data-testid="navTabs"
                >
                  <div
                    class="accordion"
                  >
                    <div
                      class="accordion-item"
                    >
                      <h2
                        class="accordion-header"
                      >
                        <button
                          aria-expanded="false"
                          class="accordion-button collapsed"
                          type="button"
                        >
                          <h3
                            class="text-uppercase"
                          >
                            fan-art
                          </h3>
                        </button>
                      </h2>
                      <div
                        class="accordion-collapse collapse"
                      >
                        <div
                          class="text-center accordion-body"
                        >
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #MOR
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #JJK
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #DVD
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #Golden Kamuy
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #HP
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #Others
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="accordion-item"
                    >
                      <h2
                        class="accordion-header"
                      >
                        <button
                          aria-expanded="false"
                          class="accordion-button collapsed"
                          type="button"
                        >
                          <h3
                            class="text-uppercase"
                          >
                            illustration
                          </h3>
                        </button>
                      </h2>
                      <div
                        class="accordion-collapse collapse"
                      >
                        <div
                          class="text-center accordion-body"
                        />
                      </div>
                    </div>
                    <div
                      class="accordion-item"
                    >
                      <h2
                        class="accordion-header"
                      >
                        <button
                          aria-expanded="false"
                          class="accordion-button collapsed"
                          type="button"
                        >
                          <h3
                            class="text-uppercase"
                          >
                            design
                          </h3>
                        </button>
                      </h2>
                      <div
                        class="accordion-collapse collapse"
                      >
                        <div
                          class="text-center accordion-body"
                        >
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #Book cover
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #T-shirt
                          </button>
                          <button
                            class="
        hvr-underline-from-center
        base
        
        my-2
      "
                          >
                            #Poster
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-5"
          data-testid="picGrid"
        >
          <div
            class="row g-3 my-3 mx-sm-3"
          >
            <div
              class="col"
            >
              <div>
                <div
                  class="react-reveal"
                  style="opacity: 0;"
                >
                   
                  <article
                    class="shadow-btn card mb-3"
                  >
                    <img
                      alt="Title"
                      src="emilywang.jpg"
                    />
                    <div
                      class="overlay"
                      data-longpress="false"
                    >
                      <div
                        class="p-3 text"
                      >
                        <h4>
                          Title
                        </h4>
                        <p
                          class="subtitle"
                        >
                          this is a test
                        </p>
                        <div
                          class="text-wrap"
                        >
                          <small
                            class="me-1"
                          >
                            #
                            tag1
                          </small>
                          <small
                            class="me-1"
                          >
                            #
                            tag2
                          </small>
                          <small
                            class="me-1"
                          >
                            #
                            tag3
                          </small>
                        </div>
                      </div>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="btn btn-primary btn-lg rounded-pill border-0 position-fixed end-0 top-0 me-3 mt-5 lh-1 shadow-btn"
          style="z-index: 1010;"
        >
          <h5
            class="text-uppercase"
          >
            :
          </h5>
        </button>
      </div>
    </div>
  </div>
</div>
`;
